### PR TITLE
Manually set Bundler root

### PR DESCRIPTION
**Pull request by gotjosh**
When the Bundler root is not set to the working directory, it attempts to
add the fully expanded path to the `Gemfile` for gems that are path based.
This resulted in an erroneous path being added to the updated `Gemfile.lock`
due to the temporary working directory not being on the original
repository.

That means that if my gem had a path of:
```YAML
PATH
  remote: path/to/my_library
```

it would output:

```YAML
PATH
  remote: tmp/dependabot_20170907-3-5asdast/path/to/my_library
```

### CODE REVIEW

**Comment by gotjosh** [comment]
@greysteil not sure why the CI is not triggering, is there something I should do?

File: spec/dependabot/shared_helpers_spec.rb [hunk 1]
```diff
@@ -3,6 +3,22 @@
 require "dependabot/shared_helpers"
 
 RSpec.describe Dependabot::SharedHelpers do
+  describe ".in_a_temporary_directory" do
```
Comment (gotjosh) [inline spec/dependabot/shared_helpers_spec.rb]: Sneaked in a small tweak to the temp dir creation helper, happy to pull that out as part of another PR if you'd not like it to go in that direction.

**Comment by greysteil** [comment]
Looks like it was a CircleCI setting - should build if you debase...

**Comment by gotjosh** [comment]
⬆️ 👌

**Comment by greysteil** [comment]
Looks great, thanks @gotjosh!
